tests/GamepadService-spec.js

```diff
@@ -112,3 +114,84 @@ describe("Consumer", () => {
     expect(consumer.hasGesture).toBe(false);
   });
 });
+
+describe("GamingDevice", () => {
+  const gamingDevices = document.createElement("div");
+  document.body.append(gamingDevices);
+
+  beforeEach(() => {
+    gamingDevices.innerHTML = "";
+  });
+
+  afterAll(() => {
+    gamingDevices.remove();
+  });
+
+  it("can be constructed with default attributes ", () => {
+    const gamingDevice = new GamingDevice();
+
+    expect(gamingDevice).toBeTruthy();
+
+    expect(gamingDevice).toBeTruthy();
+    expect(Number.isInteger(gamingDevice.id)).toBe(true);
+    expect(gamingDevice.connected).toBe(false);
+    expect(gamingDevice.buttons.length).toEqual(18);
+  });
+
+  it("has the correct number of buttons when number of buttons is specified", () => {
+    const gamingDevice = new GamingDevice(10);
+
+    expect(gamingDevice).toBeTruthy();
+
+    expect(gamingDevice).toBeTruthy();
+    expect(Number.isInteger(gamingDevice.id)).toBe(true);
+    expect(gamingDevice.connected).toBe(false);
+    expect(gamingDevice.buttons.length).toEqual(10);
+  });
+
+  it("can be constructed with default attributes and contains the correct html elements", () => {
+    const gamepadDevice = GamingDevice.attachGamingDevice(gamingDevices);
+
+    expect(gamepadDevice.parentElement).toBe(gamingDevices);
+
+    expect(gamepadDevice).toBeTruthy();
+    expect(Number.isInteger(gamepadDevice.id)).toBe(true);
+    expect(gamepadDevice.connected).toBe(false);
+    expect(gamepadDevice.buttons.length).toEqual(18);
+    expect(gamepadDevice.children[1].childElementCount).toEqual(18);
+  });
+
+  it("changes its connected status to true when any one of its buttons is clicked", () => {
```

Comment:
As above... only a particular button should be designated the "on" button. On the Playstation controller, that's: 
![Screenshot 2019-12-30 13 20 11](https://user-images.githubusercontent.com/870154/71565906-1ed0fe80-2b07-11ea-970a-f4931bfd123b.png)